### PR TITLE
Agile process guide

### PR DESCRIPTION
**Pull request by fperez**
Add the agile process guide that we had created for the SF meeting, including a new section describing how to propose new sprints with a JIRA-based workflow.

Will ping the list about this...

### CODE REVIEW

File: Agile_process.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+meeting. During this meeting the selected representative from each
+Sprint will meet with the Production Lead, LRM and Box Leads. At this
+meeting both direct and indirect impacts on Box Areas will be discussed
+and activity from the Sprints can be reviewed to ensure that design and
+implementation choices align with the broader project principles and
+goals.
+
+## Reporting and Tracking
+
+The priority list will be maintained and publicly visible. As Campaigns
```
Comment (dangunter) [inline Agile_process.md]: add a forward reference to Sprint Proposals

File: Agile_process.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+### Evaluation of sprint proposals
+
+Once a sprint/campaign proposal is in, there may be discussion and questions about it directly on the JIRA ticket.  The members of the KBase Executive Committee, who ultimately will make the decisions on which sprints to put into the implementation pipeline, will be added as watchers to the ticket so they have a chance to comment.
+
+Then, during the next maintenance week, the EC will include in its evaluation and re-prioritization decisions (that normally consider the outcome of the just-finished sprint cycle) any new proposals that have arrived.  This will allow the EC to weigh these proposals in the context of the outcomes and feedback of the recently concluded sprint cycle.
+
+The EC can decide to assign the proposals either to immediate execution in the next cycle, to be put in the queue for a later cycle, or it can decline the proposal.
+
+If a proposal is accepted, it will be given a sprint label and an epic/theme in JIRA so it can be sorted accordingly, and it will be re-assigned to its ultimate product owner.
```
Comment (dangunter) [inline Agile_process.md]: I think we need to be more prescriptive here. Every accepted proposal results in an issue of type Sprint (we really need this type, imho, not just some random labels). If it is expected to take 2+ sprints, an Epic may also be created. The title of the sprint will simply summarize the task and not include the words "sprint", "campaign", or "epic" (unless they refer to some past activity).  Since sprints tend to happen "in" an area, or at most two, the affected boxes can be placed in parentheses at the end. For example, the current title "Euk support sprint 3" would instead be "Euk support (Data)", although really this was an [outdated] Epic title and a much better one for this particular sprint would be "Data API integration with landing pages and narrative (Data)". This may sound like nitpicking, but I think clarity here is important. And it really isn't very hard to do...

Comment (nlharris) [inline Agile_process.md]: "Every accepted proposal results in an issue of type Sprint (we really need this type, imho, not just some random labels)."

In JIRA, we decided to use the built-in JIRA type "Epic", which is the JIRA/Agile term for campaigns. I could have created a new issue type "campaign" (and/or "sprint") but I used JIRA's built-in type, Epic, so that when we start using the JIRA Agile module, it works as intended, which it won't do if we make up our own issue types.
This was all discussed in email in late July.

File: Agile_process.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+
+## Reporting and Tracking
+
+The priority list will be maintained and publicly visible. As Campaigns
+are selected for a Sprint Cycle and the planning is completed, the team
+will create and assign JIRA tickets to track tasks.
+
+At the end of each Sprint, the teams will create a formal summary and
+present the results of their Retrospective on the Monday Report Out
+Meeting. These results may also include lessons learned for improving
```
Comment (dangunter) [inline Agile_process.md]: "This summary should be linked into the original Sprint issue that was created when the Sprint was approved."

File: Agile_process.md (hunk 2 above)
Comment (dangunter) [inline Agile_process.md]: An alternate plan would be to make everything an Epic as it is now, but use an integer for the Sprint field (1, 2,3, ..). Right now we're using a date, which is redundant if we used the Due Date field (and not as powerful). If we number the sprints and put the sprint start/end in the dates, then we capture all the information directly. As it is, the position of the sprint relative to others in the epic needs to be inferred by looking at all the tags in the Sprint field. Even if none of these suggestions are taken up, I think the important point is to **be explicit about how it should be done, in this document**.

Comment (scanon) [inline Agile_process.md]: Most of this happens after the proposal has been accepted and that process will be managed by myself, Fernando, or Nomi.  So I don't think we have to address this in this document.  We just need to make sure that the Jira ticket for the proposal gets put in a category so that we don't loose it.  The "Proposed Sprint" category should take care of that.

File: Agile_process.md (hunk 1 above)
Comment (fperez) [inline Agile_process.md]: done!

File: Agile_process.md (hunk 3 above)
Comment (fperez) [inline Agile_process.md]: Done, thanks!

File: Agile_process.md (hunk 2 above)
Comment (fperez) [inline Agile_process.md]: Still, we can refine it here, the point of a good PR discussion is actually to have a chance to iterate over our processes as we use them, and improve them if we see a chance to do so...

Given that for the Issue type, JIRA's agile plugin automatically creates "Epic" as a type, we might as well go with that, as @nlharris says:

image

But after that, we do have in addition, two separate label fields, 'Epic/Theme' label, _and_ 'Sprint' label.

So my question is really for @scanon: after running a few cycles, do you think we really need both types of labels?  Initially we figured we'd want to be able to label each campaign, and also be able to label by sprint as well.

But honestly, I'm starting to worry that it's a proliferation of too many management knobs, and too much information to track.

The way these labels can be used is only by _exact_ match. On the other hand, they represent a separate search field in a JQL (the "JIRA SQL" pseudo-language) query, such as this one that searches for the [sprint label 'euk-backlog'](https://atlassian.kbase.us/browse/NAR-812?jql=Sprint%20%3D%20euk-backlog).

What do people think about this?

**Comment by mlhenderson** [comment]
In order for the review process to work effectively, clear goals need to be established and the "customers" (Executive committee) need to be engaged in the process.  The Executive committee needs to define clear objectives for the organization that would then provide an objective measure for evaluating the fitness of campaigns or sprints and their progress.  Is there a document laying out the clear 3 month, 6 month, 12 month, 18 month priorities and objectives for KBase as written and decided on by the Executive committee?

**Comment by mlhenderson** [comment]
How does this process work for architectural or actual substantive changes to how the system works?  This process seems to create high pressure on choosing short objectives.  However, critical tasks for the organization such as resolving technical debt, for instance, can easily be a longer term objective that may take many iterations and sprint cycles.  Would this activity then be required to take place outside of the sprint cycles?  How is work outside of the sprint cycles reported, managed, or decided on?  Does the Executive committee review work that is not part of the sprint cycles?

**Comment by mlhenderson** [comment]
"Then, during the next maintenance week, the EC will include in its evaluation
and re-prioritization decisions (that normally consider the outcome of the
just-finished sprint cycle) any new proposals that have arrived.  This will
allow the EC to weigh these proposals in the context of the outcomes and
feedback of the recently concluded sprint cycle."

Are the EC meetings, agenda, and outcomes required to be published?  Is the EC required to meet every maintenance week regardless of the number of participants?  What is the minimum number of participants required for the EC to make these decisions?

**Comment by mlhenderson** [comment]
"The members of the KBase
Executive Committee, who ultimately will make the decisions on which sprints to
put into the implementation pipeline, will be added as watchers to the ticket
so they have a chance to comment."

What role, if any, do the box leads play in any decision process regarding sprints?

**Comment by fperez** [comment]
Not necessarily, a multi-sprint campaign can also be proposed, and I think
we should try to work with that model for simplicity.  The specifics of
technical debt can be identified into atomic chunks as part of a long-term
campaign.

On Wed, Sep 30, 2015 at 2:58 PM, Matt Henderson notifications@github.com
wrote:

> How does this process work for architectural or actual substantive changes
> to how the system works? This process seems to create high pressure on
> choosing short objectives. However, critical tasks for the organization
> such as resolving technical debt, for instance, can easily be a longer term
> objective that may take many iterations and sprint cycles. Would this
> activity then be required to take place outside of the sprint cycles? How
> is work outside of the sprint cycles reported, managed, or decided on? Does
> the Executive committee review work that is not part of the sprint cycles?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/kbase/project_guides/pull/55#issuecomment-144553897.

## 

Fernando Perez (@fperez_org; http://fperez.org)
fperez.net-at-gmail: mailing lists only (I ignore this when swamped!)
fernando.perez-at-berkeley: contact me here for any direct mail

**Comment by fperez** [comment]
On Wed, Sep 30, 2015 at 3:09 PM, Matt Henderson notifications@github.com
wrote:

> "Then, during the next maintenance week, the EC will include in its
> evaluation
> and re-prioritization decisions (that normally consider the outcome of the
> just-finished sprint cycle) any new proposals that have arrived. This will
> allow the EC to weigh these proposals in the context of the outcomes and
> feedback of the recently concluded sprint cycle."
> 
> Are the EC meetings, agenda, and outcomes required to be published?
> 
> Not in general, since other topics may be covered that can't be public, but
> we should publish this part of the outcomes.
> 
> Is the EC required to meet every maintenance week regardless of the number
> of participants? What is the minimum number of participants required for
> the EC to make these decisions?
> 
> This is something that we want feedback on, ultimately these decisions are
> up to the project PIs, who will be held responsible for project
> deliverables by DOE.  We may need the PIs to make these decisions over
> email, since getting the entire EC on the phone every three weeks is
> unrealistic.

But feedback is welcome, and we may tweak this over time based on
experience.

**Comment by fperez** [comment]
Great feedback, thanks! Do you think I should put some of the information from my replies in the document itself and update the PR  or was it mostly for discussion?

**Comment by mlhenderson** [comment]
"The EC and PI will have opportunities to
adjust the priorities during the Sprint-Free week and quarterly Project
Planning Meetings."

"However, Campaigns shouldn’t exceed four Sprints. This is to avoid
having deliverables that take too long to provide a finished product
which delays feedback."

But we're reviewing every 2 weeks, so isn't that feedback actually more
frequent?

**Comment by mlhenderson** [comment]
How is work outside of the sprint cycles reported, managed, or decided on? 
Does the Executive committee review work that is not part of the sprint cycles?

Should that be in a separate document, or does that belong here?

**Comment by mlhenderson** [comment]
I think it would be useful to update the document for clarification.

**Comment by mlhenderson** [comment]
How do we get finer granularity?  This seems critically necessary to have
clear objectives and goals that we can plan against.

**Comment by fperez** [comment]
On Wed, Sep 30, 2015 at 6:00 PM, Matt Henderson notifications@github.com
wrote:

> "The EC and PI will have opportunities to
> adjust the priorities during the Sprint-Free week and quarterly Project
> Planning Meetings."
> 
> "However, Campaigns shouldn’t exceed four Sprints. This is to avoid
> having deliverables that take too long to provide a finished product
> which delays feedback."
> 
> But we're reviewing every 2 weeks, so isn't that feedback actually more
> frequent?

Yes, perhaps we should change that language and clarify, for cases such as
testing improvement, techincal debt payoff, etc, that are sort of
'ongoing'.  They can still have concrete outcomes in each cycle, but they
can technically run for many cycles.

I didn't write that original part, actually. @scanon, how did you guys
foresee representing longer-term objectives that might be fairly specific
but still complex and multi-cycle?

**Comment by mlhenderson** [comment]
Other people should definitely comment, but I would strongly encourage the
EC to do whatever is necessary to meet and make decisions, whether by email
or not.  This process is not going to work if the EC doesn't participate.

**Comment by fperez** [comment]
On Wed, Sep 30, 2015 at 6:21 PM, Matt Henderson notifications@github.com
wrote:

> How do we get finer granularity? This seems critically necessary to have
> clear objectives and goals that we can plan against.

That is effectively a request for the EC to produce such a document on a
finer scale :)

Actually, the milestones are laid out in a quarterly manner, so in
principle they somewhat fit that granularity.

But the problem is that they aren't dynamic, they got written once.

What you are requesting is effectively that the EC provide a dynamically
updated version of milestones/priorities with roughly ~ quarterly
granularity.  Not a bad request, I think, but not something that currently
exists.

**Comment by dangunter** [comment]
The document currently ends with, "If a proposal is accepted, it will be given a sprint label and an epic/theme in JIRA so it can be sorted accordingly, and it will be re-assigned to its ultimate product owner."

We should also summarize what happens to a proposal that is rejected. First, it should be   spelled out what "reject" means. I think there can be different types of rejection, e.g.:
- REJECT: As written, this proposal is not going to happen. Need to say why, of course.
- POSTPONE: Makes sense, but we have other priorities in the foreseeable future. Give these priorities.
- RECONSIDER: We foresee doing this, but it needs to wait for N sprint cycles to be re-evaluated. Be specific about what is blocking and how many cycles.
- NOT DISCUSSED: We didn't get around to talking about it; we will get to it next time. Not ideal but it's going to happen.

Second, I think the EC should be required to put the rejection details in JIRA and communicate directly with the proposer via e-mail that this happened, so there is no confusion or feeling of "limbo" for a proposed sprint. (Internally, the EC should designate someone for this role)

**Comment by dangunter** [comment]
Hmm, sorry I see there was some categorization of rejection type in the text already. I essentially split the "decline the proposal" into REJECT and POSTPONE. This is a minor point, though I of course still tend to prefer my own breakdown.

**Comment by aparkin** [comment]
I think this is fine, though the EC is large so SOMEONE has to have the responsibility to respond. WHO on the EC is assigned the sprints? Is it me? Fernando? A fake user called "Executive Committee". I see none of these in JIRA now.

**Comment by scanon** [comment]
I will handle it or delegate someone.

**Comment by aparkin** [comment]
@scanon, great! Still that means compiling feedback from the executive. 
Also -- I know we are focused on production sprints but we need to have similar feedback mechanisms in place for SE, UE, AD, and AS. And it would be good to ensure that results are broadcasted widely after they are decided.

**Comment by scanon** [comment]
Good point.  I was thinking mainly of Production.

**Comment by scanon** [comment]
I think we should merge this now.  We can always refine it later.

**Comment by mlhenderson** [comment]
I would like it to be noted that I have unanswered questions here that have not been addressed.

**Comment by scanon** [comment]
Are their some specific ones you can repeat?

Keep in mind that this document is meant to capture the agile process not cover all of the criteria for how we make a selection.  The EC will meet regularly and review things, but the Production Lead and LRM have the latitude to adjust things based on the situation on the the ground.  So all work in the project won't stop because the EC doesn't meet.  The non-agile parts of the maintenance week aren't covered by this.  I would expect  it to be managed with jira tickets at this point.

**Comment by mlhenderson** [comment]
We should not be merging process guide documents without reading the comments.  My concern is that once this document is merged, everything I put here goes to the "forget about it" bin.  If this is the place where we should be voicing concerns, I expect those to be heard loud and clear and not be ignored.

**Comment by scanon** [comment]
I believe all the comments have been responded to, but it is a long thread and a little hard to grok.  Can you repeat any questions you feel haven't been addressed?

**Comment by mlhenderson** [comment]
This is a very bad indicator for this process, but yes I can repeat my questions.

**Comment by scanon** [comment]
> "However, Campaigns shouldn’t exceed four Sprints. This is to avoid
> having deliverables that take too long to provide a finished product
> which delays feedback."
> 
> But we're reviewing every 2 weeks, so isn't that feedback actually more
> frequent?
> 
> I didn't write that original part, actually. @scanon, how did you guys
> foresee representing longer-term objectives that might be fairly specific
> but still complex and multi-cycle?

I put that in there.  The risk of having something that spans too many sprints is it becomes very difficult to judge the progress.  I don't think the sprint reviews alone are enough to guard against that.  The tendency will be to presume that some task will get done in a subsequent sprint and I think it invites scope bloat.  I would prefer that we limit the campaigns to four sprints (I actually think it should only be two or three).  If we have something that is bigger than that, we split it up. The scrum process talks about this with story points.  They generally will break up a story if it exceeds some threshold for what can be done in a single sprint.

Also, my hope is that once we green light a campaign, we keep that going until it is finished.  So if we have something that encompasses too many sequential sprints, it impacts our ability to context switch.

**Comment by samseaver** [comment]
I'm inclined to agree with Shane on the issue of scope bloat, I generally employ beer-goggles when anticipating what can be done in any timeframe and invariably, for whatever reason, I find I can't meet the original aims within two weeks and so on. As such, I believe that we should keep individual campaigns simple.

I wonder if we would restrict campaigns to 2 or 3 sprints, but, in anticipation of Murphey and his darn law, intentionally allow time for a 3rd or 4th "clean-up" sprint.

**Comment by scanon** [comment]
Matt: I should explain that I'm not trying to cut off the conversation.  It is just we don't currently have any public version of the Agile doc (including how to propose new sprints).  So I'm having to tell folks to look at a PR.  I was hoping to get some version in place so we can make further edits from that.

**Comment by mlhenderson** [comment]
If we merge the document now, what is the process for addressing concerns I have already raised?

**Comment by scanon** [comment]
I'm not sure of the best way. Perhaps you could submit a PR with any specific suggestions for changes or a place holder PR and add the comments you think still need to be addressed.

**Comment by fperez** [comment]
I'm working right now on an update to this PR that will hopefully address some of the feedback so far.  I will then try to leave a summary of the open questions, clearly labeled.

If we agree that these are open questions, we can go ahead and merge this so we have an official version of the doc, and I propose to just file a JIRA issue with the remaining questions (some, like asking the EC for certain changes, isn't a matter to be fixed in a PR here).

How does that sound?

**Comment by cshenry** [comment]
You know, I read this through, and I understand Matt's issues. 

Speaking as a member of the EC, we currently meet for an hour about once a month. Often, a few people are absent due to conflicting meetings/schedules/etc. And of course, we've been having periodic face-to-face meetings... most recently at Berkley, and very soon in Chicago in November.

I have to say, at the month calls, the primary topic of discussion typically focuses on logistics, feedback from DOE, conferences.... essentially, day-to-day management business for running a project as large as KBase. What this meeting really isn't... is a discussion of technical and design goals for KBase. We are lucky if that is discussed for 5 minutes. 

I _don't_ think all design plans should come from the top down. I think this would be a huge mistake. Many on the EC are busy being PIs and don't even have a complete understanding of technical details of what is going on in the trenches. If we're expecting solid technical advice to come down from the EC, I would argue this is a massive mistake. Look for very broad directional pings from the EC... things like: "In the next six months, we want to be able to annotate metagenomes! Make it so!"... things like, what's written in the grant. If you're expecting more, this will fail. 

The box leads, as a part of the sprint process itself, in the long meetings we hold where the sprint progress is discussed, this is where the week-to-week nitty gritty technical decisions need to be made. Document the decisions, then pass them up to the EC, and we can review and give feedback. But you... the leads in the tenches MUST make the decisions. You make the calls. Tell us what you think. You're the one writing code and dealing with bugs and figuring out what does and does not work. Make a call! Lead. I don't think we should expect anymore more from the EC than broad directives of priorities and a "greenlight/redlight" response to proposed major technical decisions. 

I would strongly dissuade from EC micromanagement... or even midmanagement.

**Comment by aparkin** [comment]
I partly agree with this.

I think we still have hard decisions on priorities to be made.
I am not sure Exec committee meetings are the right place for review in any
case.

I think that a good summary of proposed sprints and their relationship to a
living roadmap would be useful and allow the EC to manage at the right
level.

On Thu, Oct 8, 2015 at 7:12 PM, cshenry notifications@github.com wrote:

> You know, I read this through, and I understand Matt's issues.
> 
> Speaking as a member of the EC, we currently meet for an hour about once a
> month. Often, a few people are absent due to conflicting
> meetings/schedules/etc. And of course, we've been having periodic
> face-to-face meetings... most recently at Berkley, and very soon in Chicago
> in November.
> 
> I have to say, at the month calls, the primary topic of discussion
> typically focuses on logistics, feedback from DOE, conferences....
> essentially, day-to-day management business for running a project as large
> as KBase. What this meeting really isn't... is a discussion of technical
> and design goals for KBase. We are lucky if that is discussed for 5
> minutes.
> 
> I _don't_ think all design plans should come from the top down. I think
> this would be a huge mistake. Many on the EC are busy being PIs and don't
> even have a complete understanding of technical details of what is going on
> in the trenches. If we're expecting solid technical advice to come down
> from the EC, I would argue this is a massive mistake. Look for very broad
> directional pings from the EC... things like: "In the next six months, we
> want to be able to annotate metagenomes! Make it so!"... things like,
> what's written in the grant. If you're expecting more, this will fail.
> 
> The box leads, as a part of the sprint process itself, in the long
> meetings we hold where the sprint progress is discussed, this is where the
> week-to-week nitty gritty technical decisions need to be made. Document the
> decisions, then pass them up to the EC, and we can review and give
> feedback. But you... the leads in the tenches MUST make the decisions. You
> make the calls. Tell us what you think. You're the one writing code and
> dealing with bugs and figuring out what does and does not work. Make a
> call! Lead. I don't think we should expect anymore more from the EC than
> broad directives of priorities and a "greenlight/redlight" response to
> proposed major technical decisions.
> 
> I would strongly dissuade from EC micromanagement... or even midmanagement.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/kbase/project_guides/pull/55#issuecomment-146737512.

## 

Adam Paul Arkin

## Dean A. Richard Newton Memorial Chair

Director, Physical Biosciences Division
E.O. Lawrence Berkeley National Laboratory

Professor, Department of Bioengineering
University of California
Berkeley, CA, 94720

CEO/CSO, DOE Systems Biology Knowledgebase, http://kbase.us
Director, Berkeley Synthetic Biology Institute, http://synbio.berkeley.edu
PI and Co-Director, ENIGMA SFA, http://enigma.lbl.gov http://vimss.lbl.gov
Investigator, Energy Biosciences Institute,
http://energybiosciencesinstitute.org

Office: 512C Energy Biosciences Building (Berkeley Campus)
Mailing address:
E.O. Lawrence Berkeley National Laboratory
1 Cyclotron Road, MS 955-512L
Berkeley, CA 94720

Contact:
W: http://genomics.lbl.gov
V: 510-495-2366
C: 510-206-1389
F: 510-486-6219

Assistant:
Gwyneth A. Terry
V: 510-495-2116

## E: GATerry@lbl.gov

**Comment by cshenry** [comment]
Well, I say all this because I would like to have our team at large
(coordinated and led by box leads) prepare a list of planned campaigns and
sprints leading up to the contractors meeting. We should go into the
November EC meeting with this in hand, and hopefully our discussions at
that meeting can revolve around how we want to slightly tweak the existing
plans... rather than coming up with new plans out of whole cloth.

**Comment by cshenry** [comment]
In fact, remove the scoping to just the contractor's meeting in the above
note. I would just generally like to go into the meeting with a plan to
extends out for a full year produced by our dev team and leads... I'd
rather be focusing discussions on how priorities might need to change...
rather than having to come up with priorities out of the blue.

**Comment by fperez** [comment]
BTW, one more data point: I just got back from a workshop where I had a chance to see how the LSST software team uses the JIRA agile plugin in a project whose size/complexity is right around ours (LSST is NSF-funded, and NSF agreed to have all software development for it be agile-driven).

I was quite favorably impressed, in particular because I think the tool can actually help us with the problem of having an easy to see overview of what's in flight, what is being worked on, what the next proposals are, etc.  It won't magically eliminate the complexity of understanding dependencies between resources in KBase, but I think it can be useful.

I'll talk tomorrow to @scanon and @kkellerlbl more about it so we can turn it on soon in our instance.

**Comment by benbowen** [comment]
JGI uses the Jira Agile plugin.  I like it very much.  Kind of
puppet-master style, but very cool.

**Comment by fperez** [comment]
Good to know that, @benbowen, thanks! It did look good in what I saw, and I think we can adapt gradually to it so that it's not too jarring.

**Comment by fperez** [comment]
I agree with @cshenry regarding the potential for bottlenecking if we expect the EC to actually make detailed technical decisions.  However, I think the EC/PI team should have:
- the chance to veto things before the work starts, just in case
- the responsibility to sign-off on the work for each cycle (or sequence of cycles).

Ultimately, the PIs are the ones held responsible for the project, so they need some input into this.  But you are correct that we don't want to create a micromanagement bottleneck here.

**Comment by fperez** [comment]
@dangunter, I think that 'postpone' is basically a limbo that should really be either 'reject' or 'reconsider'... But I'll add the rest of your language, thanks!

**Comment by aparkin** [comment]
@cshenry Definitely the team should come prepared to the Nov face.face with the sprints up to the contract meeting. Ideally, we will have already agreed on most things and we are working out things that need discussion. 

The role of the PIs is all this is that they are largely the product owners of the entire system. We have priorities and needs for overall system function; we have theoretically and understanding of the needs of our stakeholders and the responsibility to deliver. So we evaluate design choices and additions in this light. There are also things we as scientists want to see accomplished in and by the system and would like to see those priorities respected as well. For the former topics at least the decisions are pretty high level-- do the proposed sprints forward the product functionality as conceived (or change it positively) and are the priorities appropriate for the overall roadmap.

Basically-- I am agreeing with what Fernando just wrote but also noting that we (or speaking for me-- I) have an agenda for the overall product-- heavily influenced by the team and certainly changeable if there is great dissent-- that is supposed to guide the overall direction, design decisions and priorities. 

In large part-- we've been together long enough that this should be pretty lightweight. As I said above, largely the EC needs to get a concise description for each campaign, the current sprint, and what it serves for the product/roadmap to make most decisions.

**Comment by fperez** [comment]
I have merged this, but I created #57 so we can address the remaining questions. I tried to summarize the key points that remained open, after hopefully adding all feedback that could be cleanly integrated.

Now at least this gives us an [official, public version of the Agile guide!](https://github.com/kbase/project_guides/blob/master/Agile_process.md).
